src/pages/ProductsPage/Products.tsx

```diff
@@ -1,28 +1,47 @@
 import "./products.scss";
-import React, { useState, useEffect } from "react";
+import React, { useEffect, useState } from "react";
+import arrow from "../../assets/Mask.svg";
+import { useNavigate, useLocation } from "react-router-dom";
+import ReactPaginate from "react-paginate";
 import Product from "../../components/ui/productCard/ProductCard";
 import Loader from "../../components/ui/loader/Loader";
-import { BreedType } from "../../types";
+import { useFetchBreedsQuery } from "../../components/ui/redux-paginstion-slice/fetch-slice";
+import { useAppDispatch } from "../../shared/utils";
+import { changePage } from "../../components/ui/redux-paginstion-slice/fetch-slice";
 
 export default function Products() {
-  const [breeds, setBreeds] = useState<BreedType[]>([]);
+  const { search } = useLocation();
+  const id = Number(search.split("=")[1]);
+  const dispatch = useAppDispatch();
+  const [page, setPage] = useState(id);
+  const { data = [] } = useFetchBreedsQuery(page);
+  let breeds = data;
+  const pageCount = 17;
+  const history = useNavigate();
+
   useEffect(() => {
-    fetch("https://api.thedogapi.com/v1/breeds?limit=25&page=0")
-      .then((res) => res.json())
-      .then((data) => setBreeds(data))
-      .catch((error) => {
-        console.error("Error fetching breed information:", error);
-      });
-  }, [breeds]);
+    window.scrollTo({
+      top: 0,
+      behavior: "smooth",
+    });
+    history(`/products?page=${page}`);
+    dispatch(changePage(page));
+  }, [dispatch, history, page]);
```

Comment:
products?page=13 in the URL but actually 14th page is highlighted in the pagination. please make it consistent